cmd/gtk/dialog_wallet_create_address.go

```diff
@@ -17,31 +16,43 @@ func createAddress(ww *widgetWallet) {
 	builder, err := gtk.BuilderNewFromString(string(uiWalletCreateAddressDialog))
 	fatalErrorCheck(err)
 
+	passwordFetcher := func(wlt *wallet.Wallet) (string, bool) {
+		if *passwordOpt != "" {
+			return *passwordOpt, true
+		}
+
+		return getWalletPassword(wlt)
+	}
+
+	password, ok := passwordFetcher(ww.model.wallet)
+	if !ok {
+		return
+	}
+
 	dlg := getDialogObj(builder, "id_dialog_wallet_create_address")
 	addressLabel := getEntryObj(builder, "id_entry_account_label")
 
 	addressTypeCombo := getComboBoxTextObj(builder, "id_combo_address_type")
-	addressTypeCombo.Append(wallet.AddressTypeBLSAccount, "Account")
+	addressTypeCombo.Append(wallet.AddressTypeEd25519Account, "ED25519 Account")
+	addressTypeCombo.Append(wallet.AddressTypeBLSAccount, "BLS Account")
 	addressTypeCombo.Append(wallet.AddressTypeValidator, "Validator")
 
 	addressTypeCombo.SetActive(0)
 
-	getButtonObj(builder, "id_button_ok").SetImage(OkIcon())
-	getButtonObj(builder, "id_button_cancel").SetImage(CancelIcon())
-
 	onOk := func() {
 		walletAddressLabel, err := addressLabel.GetText()
 		fatalErrorCheck(err)
 
 		walletAddressType := addressTypeCombo.GetActiveID()
 		fatalErrorCheck(err)
 
-		if walletAddressType == wallet.AddressTypeBLSAccount {
+		switch walletAddressType {
+		case wallet.AddressTypeEd25519Account:
+			_, err = ww.model.wallet.NewEd25519AccountAddress(walletAddressLabel, password)
```

Comment:
Here you can show the password dialog. No need to create the password fetcher callback